src/views/activity/activity.view.tsx

```diff
@@ -80,12 +88,26 @@ const Activity: FC = () => {
 
   const processFetchBridgesSuccess = useCallback(
     (bridges: Bridge[]) => {
-      callIfMounted(() => {
-        setLastLoadedItem(bridges.length);
-        setBridgeList({ status: "successful", data: bridges });
+      void getPendingBridges(bridges).then((pendingBridges) => {
+        callIfMounted(() => {
+          const filteredApiBridges = bridges.filter((apiBridge) => {
```

Comment:
You need to filter the bridges that are still `on-hold` until they are complete, as now a `PendingBridge` exists and should be displayed instead of the bridge from the api.

Reply:
I still don't get this. To my understanding, whenever a bridge becomes available in the API we need to remove its shadow copy in the local storage. It should never show repeated and the API should always be the source of truth. Maybe we can jump on a call and show me what use case is covered by this `filteredApiBridges`.